spark/dl/src/main/scala/com/intel/analytics/bigdl/optim/OptimMethod.scala

```diff
@@ -63,6 +63,17 @@ trait OptimMethod[@specialized(Float, Double) T] extends Serializable {
    * @param config a table contains the hyper parameter.
    */
   def getHyperParameter(config: Table): String = ""
+
+  protected def getRegularizer(config: Table, weights: Tensor[T], weightsSign: Tensor[T]) = {
+    val rType = config.get[String]("RegularizationType").getOrElse("L2")
+    if (rType == "L1") {
+      weightsSign.copy(weights).sign()
```

Comment:
Current DenseTensor `.sign()` implementation alter the caller in place instead of  return a new one.